daemons/fenced/fenced_commands.c

```diff
@@ -2081,42 +2081,43 @@ localhost_is_eligible(const stonith_device_t *device, const char *action,
 }
 
 static void
-can_fence_host_with_device(stonith_device_t * dev, struct device_search_s *search)
+can_fence_host_with_device(stonith_device_t *dev,
+                           struct device_search_s *search)
 {
     gboolean can = FALSE;
     const char *check_type = NULL;
-    const char *host = search->host;
+    const char *target = NULL;
     const char *alias = NULL;
+    const char *action = (search == NULL)? NULL : search->action;
 
-    CRM_LOG_ASSERT(dev != NULL);
+    CRM_CHECK((dev != NULL) && (action != NULL), goto search_report_results);
 
-    if (dev == NULL) {
-        goto search_report_results;
-    } else if (host == NULL) {
+    target = search->host;
+    if (target == NULL) {
         can = TRUE;
         goto search_report_results;
     }
 
     /* Short-circuit query if this host is not allowed to perform the action */
-    if (pcmk__str_eq(search->action, "reboot", pcmk__str_casei)) {
+    if (pcmk__str_eq(action, "reboot", pcmk__str_casei)) {
         /* A "reboot" *might* get remapped to "off" then "on", so short-circuit
```

Comment:
isn't that one of the case-in-sensitive comparisons we're removing on the way ...